### ISSUE TITLE
Allow `issuePrefixes` to be a RegEx

### ISSUE THREAD

**Issue by alfaproject** [conventional-changelog/conventional-changelog#271]
Currently, I have to set a multitude of prefixes for all my projects, and it would be best if I could just allow a RegEx:

Actual: `issuePrefixes: ['XCAF-', 'ELNEW-', 'CRES-']`
Preferred: `issuePrefixes: /[A-Z]+-/`

Would be even better if we could just define what an issue is instead of just a prefix. That would allow for any system: `/#\d+`, `/[A-Z]+-\d+/`, `/\d+\$/`, etc

**Comment by hbetts** [comment]
This request seems reasonable to me.

Would you be willing and able to submit a pull request to add this to `conventional-changelog`?

**Comment by alfaproject** [comment]
Yes, I’ll give it a shot this weekend

**Comment by donifer** [comment]
@alfaproject Do you have any progress on this?

**Comment by alfaproject** [comment]
No, I'm sorry. I've stopped using this directly, so forgot about it. ):

**Comment by gerlv** [comment]
I've checked the [issue prefix code](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-commits-parser/lib/regex.js#L29), technically you can pass valid regex as a string and it will be converted into regex.

We are using JIRA for issues, the format is `JIRA-123`, but `JIRA` part is usually project-specific, i.e. `PROJA` or `PROJB`, so passing `-i "[A-Z]+-"` matches JIRA tickets.

But, it also matches `J-I-R-A-1`, doesn't match `J-I-R-A`.

I.e. `log.txt`

```
fix: something

Closes JIRA-123
```

`conventional-commits-parser -i "[A-Z]+-" log.txt` will match `JIRA-123`:

```
$ conventional-commits-parser -i "[A-Z]+-" log.txt | jq .
[
  {
    "type": "fix",
    "scope": null,
    "subject": "something",
    "merge": null,
    "header": "fix: something",
    "body": null,
    "footer": "Closes JIRA-123",
    "notes": [],
    "references": [
      {
        "action": "Closes",
        "owner": null,
        "repository": null,
        "issue": "123",
        "raw": "JIRA-123",
        "prefix": "JIRA-"
      }
    ],
    "mentions": [],
    "revert": null
  }
]
```

**Comment by fackyhigh** [comment]
issuePrefixes are RegEx, actually: https://github.com/conventional-changelog/conventional-changelog/blob/d0e5d5926c8addba74bc962553dd8bcfba90e228/packages/conventional-commits-parser/lib/regex.js#L36

For example, if you write `--issueUrlFormat` like this: `https://jira.com/browse/{{id}}` with `--issuePrefixes "[A-Z]+-"`, with commit message "_Fix something [JIRA-123, ARIJ-321]_", you'll get such URLs: 
https://jira.com/browse/123 and https://jira.com/browse/321 that are not correct JIRA IDs.

Here is the workaround:
Commit message: "Fix something [JIRA-123, ARIJ-321]"
Flags: `--issueUrlFormat "https://jira.com/browse/{{prefix}}{{id}}" --issuePrefixes "[A-Z]+-"`
Result: https://jira.com/browse/JIRA-123 and https://jira.com/browse/ARIJ-321
